src/utils/utils.ts

```diff
@@ -31,15 +34,94 @@ export const getFilterParams = (filterString: string | null) => {
 };
 
 export const parseBytes = (bytes: number) => {
-  const prefixes = [" B", " kB", " MB", " GB", " TB", " PB", " EB", " ZB", " YB"];
+  const prefixes = [
+    " B",
+    " kB",
+    " MB",
+    " GB",
+    " TB",
+    " PB",
+    " EB",
+    " ZB",
+    " YB",
+  ];
   let parsedBytes = prefixes.flatMap((prefix, index) => {
-    let calculatedVal = bytes / Math.pow(1000, index)
+    let calculatedVal = bytes / Math.pow(1000, index);
     if (calculatedVal < 1000 && calculatedVal >= 0.1) {
-      return String(Math.round(calculatedVal * 100) / 100) + prefix
-    }
-    else return null
+      return String(Math.round(calculatedVal * 100) / 100) + prefix;
+    } else return null;
   });
-  var returnValue = parsedBytes.find((parsing) => parsing !== null)
-  if (returnValue === undefined) returnValue = String(bytes) + prefixes[0]
-  return returnValue
+  var returnValue = parsedBytes.find((parsing) => parsing !== null);
+  if (returnValue === undefined) returnValue = String(bytes) + prefixes[0];
+  return returnValue;
+};
+
+export const handleSearch = (
+  setSearchResults: Dispatch<SetStateAction<searchResponseModel | null>>,
+  filterDict: facetFilterModel[],
+  searchKeyword: string,
+  limit: number,
+  setSearchParams: any,
+  setPage: Dispatch<SetStateAction<number>>
+) => {
+  let skip = 0;
```

Comment:
The common lines between handleSearch and handleFilter can be moved to a new method. This will improve the code quality and reduce code duplication.

Reply:
Done. Fixed bug from 849 while at it.